pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -103,3 +104,17 @@ export const download = (blob: Block): string => {
 
   return url;
 };
+
+export const getBlockByUlid = (blocks: Block[], ulid: string): string => {
```

Comment:
Instead of returning the most relevant ulid, let's return a block array, with all the matching blocks.